### PR TITLE
Collateral class added.

### PR DESCRIPTION
**Pull request by Purvanshsingh**
I have added Collateral class to `NPL Vocab` & ` ApiDoc`.
All the CRUD functionalities are added to the Collateral class.

DB structure with Collateral class :
Screenshot from 2021-06-21 13-48-14

GET, PUT, & POST operations :

Screenshot from 2021-06-21 14-48-50

Screenshot from 2021-06-21 14-49-15

Screenshot from 2021-06-21 14-49-31

### CODE REVIEW

**Comment by open-risk** [comment]
Some further tests:
- To test with actual loan id's
- To test multiple collateral with same loan id
- To test cascaded delete of collateral if the borrower class to which loan (and collateral) is linked is deleted

File: creditrisk_poc/api_doc/ApiDoc.jsonld [hunk 1]
```diff
@@ -611,6 +717,103 @@
                     "required": null,
                     "writeable": false
                 },
+                {
+                    "hydra:description": "The Collateral Class",
```
Comment (Mec-iS) [inline creditrisk_poc/api_doc/ApiDoc.jsonld:721]: there shouldn't be here an `@id` or a `@type` that points to the `Collateral` class in the original ontology?
cc @chrizandr

File: creditrisk_poc/api_doc/api_docwriter.py [hunk 2]
```diff
@@ -246,4 +342,4 @@
             doc_file.write(json_doc)
             print("Your API_Doc has be successfully created.")
     except Exception as error:
-        print(error, "Occurred while saving API_Doc")
\ No newline at end of file
+        print(error, "Occurred while saving API_Doc")
```
Comment (Mec-iS) [inline creditrisk_poc/api_doc/api_docwriter.py]: please use `logging` not `print`

File: creditrisk_poc/api_doc/ApiDoc.jsonld (hunk 1 above)
Comment (Purvanshsingh) [reply creditrisk_poc/api_doc/ApiDoc.jsonld:721]: @Mec-iS Complete descriptions are not added as we are more focused on functionality as of now.

**Comment by Purvanshsingh** [comment]
@open-risk I have added the `mock_portfolio_generator.py` script.

It generates :
-  100 borrowers with no loan & no collateral
-  100 borrowers with loan & no collateral
-  100 borrowers with loan & collateral

And add them to the database via `PUT` request.

Borrower -> Loan -> Collateral are linked with the foreign key.

Tested: on deleting a borrower the respective **Loan** and the **collateral** associated with the loan_id deleted automatically.

**Review by open-risk — APPROVED** [review]

**Pull request merged**